[FIX] azure_oauth: configuration mail server Outlook. (#5909)
odoo/documentation: content/administration/maintain/azure_oauth.rst

Finally, click :guilabel:`Test Connection`. A confirmation message should appear. The Odoo database
can now send safe, secure emails through Microsoft Outlook using OAuth authentication.

.. _azure_oauth/notifications:

Configuration with a single outgoing mail server
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Configuring a single outgoing server is the simplest configuration available for Microsoft Azure
and it doesn't require extensive access rights for the users in the database.

A generic email address would be used to send emails for all users within the database. For example
it could be structured with a `notifications` alias (`notifications@example.com`) or `contact` alias

> (contact@example.com). This address must be set as the :guilabel:`FROM Filtering` on the server.

Reviewer: To avoid the example email formatting as a _mailto:_ link, escape the domain using `\` if this is the desired formatting output 🙂
```suggestion
(contact\@example.com). This address must be set as the :guilabel:`FROM Filtering` on the server.
```

This address must also match the `{mail.default.from}@{mail.catchall.domain}` key combination in the
system parameters.

> .. seealso::
>    For more information on the from filter visit :ref:`this documentation
>    <email_communication/default>`.

Reviewer: I think this may be gray area, but I prefer to stay away from ambiguous link titles (similar to alt text on images). I would rather the link title to be descriptive. By removing the label we can just use the header the anchor is linking to, or provide a more descriptive label than "this documentation". What do you think? I will suggest this for all ambiguous link labels 😉
```suggestion
.. seealso::
   For more information on the from filter visit: :ref:`email_communication/default`.
```

Reviewer 2: I prefer this too.

.. note::

>    The :guilabel:`System Parameters` can be accessed by activating
>    :doc:`../../applications/general/developer_mode` in the :menuselection:`Settings --> Technical
>    --> Parameters --> System Parameters` menu.

Reviewer: Can use the :ref: here
```suggestion
   The :guilabel:`System Parameters` can be accessed by activating :ref:`developer-mode` in the
   :menuselection:`Settings --> Technical --> Parameters --> System Parameters` menu.
```

When using this configuration, every email that is sent from the database will use the address of
the configured `notification` mailbox. However it should be noted that the name of the sender will
appear but their email address will change:

.. image:: azure_oauth/from-name-remain.png
   :align: center
   :alt: Name from real sender with static email.

.. example::
   Single outgoing mail server configuration:

   - Outgoing mail server **username** (login) = `notifications@example.com`

>    - Outgoing mail server  :guilabel:`FROM Filtering` = `notifications@example.com`

Reviewer: Remove extra space
```suggestion
   - Outgoing mail server :guilabel:`FROM Filtering` = `notifications@example.com`
```

   - `mail.catchall.domain` in system parameters = `example.com`
   - `mail.default.from` in system parameters = `notifications`

User-specific (multiple user) configuration
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In addition to a generic email server, individual email servers can be set up for users in a
database. These email addresses must be set as the :guilabel:`FROM Filtering` on each individual
server for this configuration to work.

This configuration is the more difficult of the two Microsoft Azure configurations, in that it
requires all users configured with email servers to have access rights to settings in order to
establish a connection to the email server.

Setup
*****

Each user should have a separate email server set up. The :guilabel:`FROM Filtering` should be set
so that only the user's email is sent from that server. In other words, only a user with an email
address that matches the set :guilabel:`FROM Filtering` is able to use this server.

.. seealso::

>    For more information on the from filter visit :ref:`this documentation
>    <email_communication/default>`.

Reviewer: Ambiguous link label
```suggestion
   For more information on the from filter visit: :ref:`email_communication/default`.
```

> A fallback server must be setup to allow for the sending of :guilabel:`notifications`. :ref:`See
> this documentation <azure_oauth/notifications>`. The :guilabel:`FROM Filtering` for this server
> should have the value of the `{mail.default.from}@{mail.catchall.domain}`.

Reviewer: Ambiguous link label
```suggestion
A :ref:`fallback server <azure_oauth/notifications>` must be setup to allow for the sending of
:guilabel:`notifications`. The :guilabel:`FROM Filtering` for this server should have the value of
the `{mail.default.from}@{mail.catchall.domain}`.
```

.. note::

>    The :guilabel:`System Parameters` can be accessed by activating
>    :doc:`../../applications/general/developer_mode` in the :menuselection:`Settings --> Technical
>    --> Parameters --> System Parameters` menu.

Reviewer: Can use the :ref: here
```suggestion
   The :guilabel:`System Parameters` can be accessed by activating :ref:`developer-mode` in the
   :menuselection:`Settings --> Technical --> Parameters --> System Parameters` menu.
```

.. important::
   The configuration for this transactional email server can work alongside an outgoing mass-mailing
   email server. The :guilabel:`FROM Filtering` for the mass-mailing email server can remain empty,

>    but it's require to be added in the Settings of the email marketing application.

Reviewer: Capitalization and app name formatting
```suggestion
   but it's require to be added in the settings of the *Email Marketing* application.
```

   .. seealso::

>       For more information on setting the mass-mailing email server visit :ref:`this documentation
>       <email_communication/mass_mails>`.

Reviewer: Ambiguous link label
```suggestion
      For more information on setting the mass-mailing email server visit:
      :ref:`email_communication/mass_mails`.
```

.. example::
   Multiple user outgoing mail server configuration:

   - Notifications mailbox
      - Outgoing mail server #1 **username** (login) = `notifications@example.com`

>       - Outgoing mail server #1 :guilabel:`FROM Filtering` = `notifications@example.com`

Author: Can you put this Notifications mailbox AFTER the 2 User ?
- Visually speaking it makes more sense (this is a fallback)
- it matches the explanation given previously (first configure 1 OMS / user then a fallback (...) )

Reviewer 3: Added this in, resolved issue.

Finally, click :guilabel:`Test Connection`. A confirmation message should appear. The Odoo database
can now send safe, secure emails through Microsoft Outlook using OAuth authentication.

.. _azure_oauth/notifications:

Configuration with a single outgoing mail server
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Configuring a single outgoing server is the simplest configuration available for Microsoft Azure
and it doesn't require extensive access rights for the users in the database.

A generic email address would be used to send emails for all users within the database. For example
it could be structured with a `notifications` alias (`notifications@example.com`) or `info` alias

> (info@example.com). This address must be set as the :guilabel:`FROM Filtering` on the server. This

Reviewer 3: Noted, I've corrected this.

address must also match the `{mail.default.from}@{mail.catchall.domain}` key combination in the
system parameters.

.. seealso::
   For more information on the from filter visit :ref:`this documentation
   <email_communication/default>`.

.. note::
   The :guilabel:`System Parameters` can be accessed by activating
   :doc:`../../applications/general/developer_mode` in the :menuselection:`Settings --> Technical
   --> Parameters --> System Parameters` menu.

When using this configuration, every email that is sent from the database will use the address of
the configured `notification` mailbox. However it should be noted that the name of the sender will
appear but their email address will change:

.. image:: azure_oauth/from-name-remain.png
   :align: center
   :alt: Name from real sender with static email.

.. example::
   Single outgoing mail server configuration:

   - Outgoing mail server login = `notifications@example.com`
   - Outgoing mail server From filter = `notifications@example.com`
   - `mail.catchall.domain` in system parameters = `example.com`
   - `mail.default.from` in system parameters = `notifications`

Multiple user configuration

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer 3: I've added this in, thanks for the suggestion.

In addition to a generic email server, individual email servers can be set up for users in a
database. These email addresses must be set as the :guilabel:`FROM Filtering` on each individual
server for this configuration to work.

This configuration is the more difficult of the two Microsoft Azure configurations, in that it
requires all users configured with email servers to have access rights to settings to establish a

> connection to the email server.

Reviewer 3: I have clarified this sentence a bit. I am leaving the work-around out for now, since this isn't standard anymore. 

Let me know what you think. 

Thanks,
Tim

   email server. The :guilabel:`FROM Filtering` for the mass-mailing email server can remain empty,
   but it's require to be added in the Settings of the email marketing application.

   .. seealso::
      For more information on setting the mass-mailing email server visit :ref:`this documentation
      <email_communication/mass_mails>`.

.. example::
   Multiple user outgoing mail server configuration:

   - Outgoing mail server #1 login = `notifications@example.com`
   - Outgoing mail server #1 From filter = `notifications@example.com`
   - Outgoing mail server #2 login = `john@example.com`
   - Outgoing mail server #2 From filter = `john@example.com`
   - Outgoing mail server #3 login = `jane@example.com`
   - Outgoing mail server #3 From filter = `jane@example.com`
   - `mail.catchall.domain` in system parameters = `example.com`
   - `mail.default.from` in system parameters = `notifications`

.. warning::
   Misconfiguration can occur when the `mail.default.from` system parameter is set to use a full
   email address that doesn't belong to the `mail.catchall.domain` system parameter. The

>    `mail.catchall.domain` system parameter should be edited to match the From Filter.

Reviewer 3: How would you configure this?

>    `mail.catchall.domain` system parameter should be edited to match the From Filter.

Author: This is not a reply, it's an explanation of no configuration to do.

As long as we don't do a FAQ, I would not mention it.

>    `mail.catchall.domain` system parameter should be edited to match the From Filter.

Reviewer 3: Removed this warning block per @Abridbus

:+1:
